app/src/main/java/com/github/se/eduverse/viewmodel/FileViewModel.kt

```diff
@@ -71,4 +77,54 @@ class FileViewModel(val fileRepository: FileRepository) {
     _validNewFile.value = false
     _newFile.value = null
   }
+
+  /**
+   * Open a pdf file using an specialized app on the device
+   *
+   * @param fileId the id of the file as stored in firestore
+   * @param context the context in which this method is used
+   */
+  fun openFile(fileId: String, context: Context) {
+    fileRepository.accessFile(
+        fileId = fileId,
+        onSuccess = { pdfRef ->
+          val localFile = File.createTempFile("tempFile", ".pdf")
```

Comment:
Temporary files you create (File.createTempFile("tempFile", ".pdf")) should be deleted after use to avoid the accumulation of unnecessary temporary files. It may be worth using a try-finally or some mechanism to ensure deletion.  

example: try {
    pdfRef.getFile(localFile).addOnSuccessListener { openPDF(localFile, context) }
} finally {
    localFile.delete()
}

Reply:
This is a good point. I first thought that createTempFile created a file which was automaticly deleted by the os when not used, but after checking it seems that was wrong. I corrected it the way you suggested.